src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageSender.cs

```diff
@@ -139,7 +139,13 @@ internal AmqpMessageSender(string entityName, MessagingEntityType entityType, Se
                     null);
             request.Map[ManagementConstants.Properties.SequenceNumbers] = new[] { sequenceNumber };
 
-            await this.ExecuteRequestResponseAsync(request);
+            var response = await this.ExecuteRequestResponseAsync(request);
+
+            // TODO: no idea what code and exception should be thrown, but the should be
+            if (response.StatusCode != AmqpResponseStatusCode.OK)
+            {
+                // throw
```

Comment:
@SeanFeldman , can you please merge with the latest pack and then throw 
throw response.ToMessagingContractException();